TCES-60 client routing

kevinle623 opened the pull request:
# Description

Implemented Client Routing

Issue # 60

## Gallary 

![image](https://github.com/uoftblueprint/tces/assets/74465053/2ad6b72e-4e4f-407a-bb09-f440b69139d8)
![image](https://github.com/uoftblueprint/tces/assets/74465053/43399a69-b67f-47cd-908f-61c3cd6dee7d)

<img width="766" alt="image" src="https://github.com/uoftblueprint/tces/assets/74465053/81e45b75-5f66-4c1b-b2d6-79861f4938e1">



## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist

- [x] My code follows the style guidelines of this project (required)
- [x] I have performed a self-review of my code (required)
- [x] I have linted and formatted my code (required)
- [x] New and existing unit tests pass locally with my changes (required)
- [x] I have commented my code, particularly in hard-to-understand areas

kevinle623 committed:
set up table pagination

kevinle623 committed:
implemented add client feature

kevinle623 committed:
added modify client feature + fixed job lead edit crash

kevinle623 committed:
Merge branch 'main' into kevinle6231/tces-60-client-routing

kevinle623 committed:
merged with master

kevinle623 committed:
fixed styling for the apply filter and reset button

kevinle623 committed:
Merge branch 'main' into kevinle6231/tces-60-client-routing

DanielDervishi commented on client/src/App.jsx:
```diff
@@ -15,6 +15,8 @@ import LoginPage from "./pages/login";
 import CreatePage from "./pages/create-user";
```
Just a few things I noticed weren't working through the web interface:
- Editing the client's status: Tried changing World Health Organization status
- Can't change the filter status
- Status should be defaulting to active: (status shouldn't be able to become unknown)
- The table currently shows "Unknown" as the owner of any client who User 2 owns
- Changing the status of the user John Doe to closed isn't reflected after submitting
- Clicking on org 6's page shows the following
<img width="1003" alt="Screenshot 2024-02-16 at 5 52 00 PM" src="https://github.com/uoftblueprint/tces/assets/58835213/2e61ddc2-5268-48b8-a2da-2217c476613b">

kevinle623 replied:
Hi Daniel, thank you so much for the catches and sharing the error prompt. I have pushed changes to fix the issues listed above. Modify status and closing should behave as normally now.

However, for the point "The table currently shows "Unknown" as the owner of any client who User 2 owns". The reason is because, we actually deleted the user with id 2 from our user table from the admin dashboard a while back so user with id 2 no longer exists. That's why if a user with a certain id can't be found, we just display it   as unknown for now. Let me know if this is fine.

DanielDervishi replied:
Hey Kevin, that sound totally reasonable, thanks for the explanation 👍

kevinle623 committed:
fixed issues

kevinle623 committed:
tweak to changed owner logic

jordanjanakievski reviewed (commented):
Everything is looking and feeling great while interacting with the pages and filters.

Just a couple of comments around some closure information and filtering

<img width="1359" alt="Screenshot 2024-02-19 at 18 39 19" src="https://github.com/uoftblueprint/tces/assets/41130718/4ed1c26a-34f8-4a8c-8ee4-9e9ed2c044db">

This may be a backend issue on its own, but when I change the status to closed, no closure date is showing up when returning and therefore a time since the closure. Maybe this needs to be extracted to another ticket.

And then obviously the 3 months since is showing up, but my guess it that is due to not having a close date.

---

Small Comments

- Unable to filter by status -> Active, R&I, Closed
- Some names are displayed as unknown

kevinle623 committed:
fixed filter and closed date stats

kevinle623 committed:
cleaned up function

kevinle623 commented:
> Everything is looking and feeling great while interacting with the pages and filters.
> 
> Just a couple of comments around some closure information and filtering
> 
> <img alt="Screenshot 2024-02-19 at 18 39 19" width="1359" src="https://private-user-images.githubusercontent.com/41130718/306093925-4ed1c26a-34f8-4a8c-8ee4-9e9ed2c044db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgzOTA4MDksIm5iZiI6MTcwODM5MDUwOSwicGF0aCI6Ii80MTEzMDcxOC8zMDYwOTM5MjUtNGVkMWMyNmEtMzRmOC00YThjLThlZTQtOWU5ZWQyYzA0NGRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIwVDAwNTUwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjMjFiY2JjYzgxY2M3MTMxN2U4MTYyZWRkZWFjNGU0NTA0NGRiNGJiMjUxYWUyYThkMDJmZjlhNjA3NjE0MzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mM43blb2n9RKEJ0r6lTcZOKcO9ccN-K-iQjoy3rwsZo">
> This may be a backend issue on its own, but when I change the status to closed, no closure date is showing up when returning and therefore a time since the closure. Maybe this needs to be extracted to another ticket.
> 
> And then obviously the 3 months since is showing up, but my guess it that is due to not having a close date.
> 
> Small Comments
> 
> * Unable to filter by status -> Active, R&I, Closed
> * Some names are displayed as unknown

Hi Jordan, thanks for your review, I have fixed the status closed logic for the closure date being missing and the time since the closure logic. Thanks for catching that.

In regards to the small comments, I have fixed the filter by status. However, for the names being displayed as unknown, the reason is because that user has been deleted from the db from the admin dashboard so because of that we can't find the user thus displaying Unknown. Hope this makes sense, thanks.

jordanjanakievski commented:
> In regards to the small comments, I have fixed the filter by status. However, for the names being displayed as unknown, the reason is because that user has been deleted from the db from the admin dashboard so because of that we can't find the user thus displaying Unknown. Hope this makes sense, thanks.

Sounds good. We have to add that logic to the deletion of users still to default all ownership and creation to admin

jordanjanakievski reviewed (dismissed):
Looks good to me! Thanks Kevin!

kevinle623 committed:
Merge branch 'main' into kevinle6231/tces-60-client-routing

kevinle623 committed:
merged with master and fixed styling

DanielDervishi commented on client/src/App.jsx:
```diff
@@ -15,6 +15,8 @@ import LoginPage from "./pages/login";
 import CreatePage from "./pages/create-user";
```
Since we changed the title of the Client table page from "My Clients" to "Clients", it also be changed here
<img width="1212" alt="Screenshot 2024-02-25 at 10 43 14 PM" src="https://github.com/uoftblueprint/tces/assets/58835213/04e950a8-8258-4547-ab69-cffb08a49c53">

kevinle623 replied:
switched, thanks!

DanielDervishi reviewed (dismissed):
Great Work Kevin! Just left one quick comment

kevinle623 committed:
switched to "Clients" heading
